bigtable/google/cloud/bigtable/table.py

```diff
@@ -276,6 +284,31 @@ def read_rows(self, start_key=None, end_key=None, limit=None,
         # We expect an iterator of `data_messages_v2_pb2.ReadRowsResponse`
         return PartialRowsData(response_iterator)
 
+    def mutate_rows(self, rows):
+        """Mutates multiple rows in bulk.
+    
+        :type rows: list
+        :param rows: List or other iterable of :class:`.DirectRow` instances.
+        
+        :rtype: list
+        :returns: A list of tuples (``MutateRowsResponse.Entry`` protobuf 
+                  corresponding to the errors, :class:`.DirectRow`) 
+        """
+        _check_rows(self.name, rows)
+        mutate_rows_request = _mutate_rows_request(self.name, rows)
```

Comment:
This is problematic because `_check_rows` will **exhaust** the passed in iterator `rows` before you iterate through it again in `_mutate_rows_request`. So this would **only** work if `rows` was something like a `list` / `tuple` / etc. that can be iterated over many times.

Suggested fix: convert `_check_rows` into a `_check_row` helper and then apply it within `_mutate_rows_request` as you iterate through `rows`.